ksqldb-engine/src/test/java/io/confluent/ksql/physical/pull/operators/KeyedTableLookupOperatorTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.physical.pull.operators;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.confluent.ksql.execution.streams.materialization.Locator.KsqlNode;
+import io.confluent.ksql.execution.streams.materialization.Locator.KsqlPartitionLocation;
+import io.confluent.ksql.execution.streams.materialization.Materialization;
+import io.confluent.ksql.execution.streams.materialization.MaterializedTable;
+import io.confluent.ksql.execution.streams.materialization.Row;
+import io.confluent.ksql.execution.streams.materialization.ks.KsLocator;
+import io.confluent.ksql.planner.plan.DataSourceNode;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.apache.kafka.connect.data.Struct;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@SuppressWarnings({"UnstableApiUsage", "unchecked"})
+@RunWith(MockitoJUnitRunner.class)
+public class KeyedTableLookupOperatorTest {
+
+  private final List<KsqlPartitionLocation> singleKeyPartitionLocations = new ArrayList<>();
+  private final List<KsqlPartitionLocation> multipleKeysPartitionLocations = new ArrayList<>();
+
+  @Mock
+  private KsqlNode node1;
+  @Mock
+  private KsqlNode node2;
+  @Mock
+  private KsqlNode node3;
+  @Mock
+  private Materialization materialization;
+  @Mock
+  private MaterializedTable nonWindowedTable;
+  @Mock
+  private DataSourceNode logicalNode;
+  @Mock
+  private Struct KEY1;
+  @Mock
+  private Struct KEY2;
+  @Mock
+  private Struct KEY3;
+  @Mock
+  private Struct KEY4;
+  @Mock
+  private Row ROW1;
+  @Mock
+  private Row ROW3;
+  @Mock
+  private Row ROW4;
+
+
+  @Before
```

Comment:
`@Before` runs before every test, not once before all the tests. If you want these to be populated once, which I think you do, it should be in the constructor